images/image-alt-text-context-form.html

```diff
@@ -0,0 +1,164 @@
+            content_style: `
+                body {
+                    min-height: 100vh;
+                    margin: 2rem;
+                }
+
+                img {
+                    max-width: 100%;
+                }
+            `,
+
+            setup: (editor) => {
+                // Instead of defining a custom icon pack, we can register individual
+                // icons using the API.
+                // https://www.tiny.cloud/docs/api/tinymce.editor.ui/tinymce.editor.ui.registry/#addicon
+                editor.ui.registry.addIcon('alt-text', '<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24"><g fill-rule="evenodd"><path fill-rule="nonzero" d="M16.144 3.372l1.286.772L7.658 20.43l-1.286-.772.614-1.024c-1.099-.627-2.13-1.451-3.089-2.436a18.159 18.159 0 0 1-2.06-2.547l-.127-.192-.223-.35-.18-.3-.14-.247a6.025 6.025 0 0 1-.054-.101 1 1 0 0 1 0-.922l.054-.101.14-.247.18-.3.223-.35.127-.192a18.159 18.159 0 0 1 2.06-2.547C6.2 5.438 8.908 4 12 4c1.19 0 2.324.213 3.4.61l.744-1.238zM12 6C9.523 6 7.289 7.187 5.33 9.198A16.178 16.178 0 0 0 3.153 12l.217.342c.04.063.083.127.128.195a16.178 16.178 0 0 0 1.832 2.265c.845.868 1.742 1.583 2.687 2.114l1.216-2.027a4 4 0 0 1 4.015-6.69l1.099-1.832A7.704 7.704 0 0 0 12 6z"/><path d="M18.007 18.002C16.204 19.272 14.199 20 12 20a9.71 9.71 0 0 1-2.231-.258l1.064-1.74h7.174zM13.279 14h8.641l-.057.085c-.465.652-.989 1.299-1.572 1.917L12.056 16l1.223-2zm2.444-4h6.2l.121.175.246.366.223.35.18.3.14.247.054.101A1 1 0 0 1 23 12h-8.5l1.223-2zm4.38-2.198l.186.198h-3.344l1.175-1.922a15.12 15.12 0 0 1 1.983 1.724z"/></g></svg>');
+
+                // Register a custom context toolbar shown when selecting any image
+                // https://www.tiny.cloud/docs/ui-components/contexttoolbar/#registeringacontexttoolbar
+                editor.ui.registry.addContextToolbar('imageContextToolbar', {
+                    predicate: function (node) {
+                        return node.nodeName.toLowerCase() === 'img'
+                    },
+                    items: 'alttext', // The context toolbar only contains our alt-text context form trigger button
+                    position: 'node',
+                    scope: 'node'
+                });
+
+                // The context form contains our alt text input field. It also manages
+                // the toolbar button that opens the context form.
+                // https://www.tiny.cloud/docs/ui-components/contextform/
+                editor.ui.registry.addContextForm('alttext', {
+                    // Define the button used in our context toolbar to open the context form
+                    launch: {
+                        type: 'contextformbutton',
+                        icon: 'alt-text',
+                        tooltip: 'Image description (Alt text)'
+                    },
+
+                    // The input field label (Currently does nothing).
+                    label: 'Image description (alt text)',
```

Comment:
The label property doesn't do anything what I can see. If that's the case, I'll write up a change that the label will show up as a placeholder on the input field.

Reply:
It's used for ARIA purposes, as it adds the `aria-label` attribute:
![image](https://user-images.githubusercontent.com/1575550/94499723-cc14a680-0240-11eb-8a04-108b65e70ac3.png)

Reply:
Nice!